docs/contributor/testing.md

```diff
@@ -0,0 +1,64 @@
+| Test execution                        | A part of the CI process                        |                                                              |                                                              |
+​
+The roles and responsibilities during the STLC:
+​
+| Role        | Responsibilities                                             | Performed by     |
+| ----------- | ------------------------------------------------------------ | ---------------- |
+| `PO`        | Define acceptance criteria for the user stories, assess risks | `PO`             |
+| `DEV`       | Implementation of tests for new functionality, an extension of the test suites, adherence to the test coverage expectations | development team |
+| `ARCHITECT` | Devise system design with a focus on testability             | team-shared role |
+| `QA`        | Define the testing coverage for each story, and ensure the test suite is delivered along with each new piece of functionality. | team-shared role |
+​
+## Testing Levels
+​
+### Functional Tests
+​
+`Unit` and `Env`-tests follow the go convention and reside next to the code they are testing. The unit tests and integration tests are a part of one test suite.
+​
+| Test suite                                                   | Testing level                         | Purpose                                                      |
+| ------------------------------------------------------------ | ------------------------------------- | ------------------------------------------------------------ |
+| Unit (located with the individual source files)              | Unit                                  | It tests the units in complete isolation. This test suite assesses the implementation correctness of the units of business logic. |
+| Env-tests (located with the individual source files)         | Integration  (low-level)              | It tests the behaviour of the Telemetry Controller in integration with a k8s API server replaced with a test double. This test suite assesses the integration correctness of the controller. |
+| [E2E](/test/e2e) | Acceptance / Integration (high-level) | It tests the usability scenarios of the Telemetry Controller in a cluster. This test suite assesses the functional correctness of the controller. |
+​
+### Non-functional Tests
+​
+| Type                                                         | Automation | Frequency                               | Results          |
+| ------------------------------------------------------------ | ---------- | --------------------------------------- | ---------------- |
+| Release testing | Manual     | Regularly before each release           | Manual tests     |
+| Performance tests | Manual     | Ad hoc on a noticeable component change | Tests repository |
```

Comment:
minor version change maybe, we have not defined explicit yet

Reply:
Will we define it soon? I'd prefer to get rid of "noticeable" because it's kinda empty in meaning, but not worth to block progress here because of that one word hidden in a table.